playwright/stream.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import io
+from typing import Dict, Optional
+
+from playwright.connection import ChannelOwner
+
+
+class Stream(ChannelOwner):
+    def __init__(
+        self, parent: ChannelOwner, type: str, guid: str, initializer: Dict
+    ) -> None:
+        super().__init__(parent, type, guid, initializer)
+
+    async def stream(self, fp: io.BytesIO, size: Optional[int]) -> io.BytesIO:
```

Comment:
I don't understand the version that returns the stream. It looks like it'll create one on line 30 and then read all content into its buffer? So there isn't much streaming going one?

Reply:
Looks like you need to subclass https://docs.python.org/3/library/io.html#io.RawIOBase in your stream instead.

Reply:
True, we should probably rename it. But in the Python world there are no streams like in Node.js, thats why I used this approach.
Could be used for writing into files or getting the content als a BytesIO stream.
Their documentation says for [io.BytesIO](https://docs.python.org/3/library/io.html#buffered-streams):
> A stream implementation using an in-memory bytes buffer. It inherits BufferedIOBase. The buffer is discarded when the close() method is called.

If we have another name we don't have documentation on the other hand, I would keep the name probably.